### PR TITLE
Adds skipBody CLI argument - Skips Request and Response Body in Try Output

### PR DESCRIPTION
**Pull request by Pagan-Idel**
- Cargo Version Upgrade
- Updates the guide to include new flag
- Adds the ability to include skipBody argument in the CLI
- Changes added to test the new argument

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -856,8 +859,8 @@ fn create_try_run_future(
          ========================================\n\
          ${response['start-line']}\n\
          ${join(response.headers_all, '\n', ': ')}\n\
-         ${if(response.body != '', '\n${response.body}', '')}\n\n`""#
-    } else {
+         ${if(response.body != '', '\n${response.body}', '')}\n`""#
+    } else if matches!(try_config.format, TryRunFormat::Json) && matches!(try_config.skip_body_on, false) {
```
Comment (tkmcmaster) [inline src/lib.rs:863]: same here. A simple boolean check doesn't need `matches!`

**Review by tkmcmaster — CHANGES_REQUESTED** [review]
We don't bump the version number every time. We'll do a pre-release version and make sure everything works and then release it later.

There's probably a simpler way than doing so many if/then/elses. Can we have a separate check that looks at skip body and sets a variable value? Then don't include that later?

Something like:

```
let body_display = if body on {
  // assign body based on json or human
} else {
  // return empty string
}
```
That way we don't have to tweak 4 different text blocks if other things change?

File: src/lib.rs [hunk 2]
```diff
@@ -845,8 +848,8 @@ fn create_try_run_future(
     debug!("create_try_run_future start");
     // create a logger for the try run
     // request.headers only logs single Accept Headers due to JSON requirements. Use headers_all instead
-    let select = if matches!(try_config.format, TryRunFormat::Human) {
-        r#""`\
+    let select = if matches!(try_config.format, TryRunFormat::Human) && matches!(try_config.skip_body_on, false){
```
[the remark this answers is not in the document]
Comment (Pagan-Idel) [reply src/lib.rs:851]: done

File: src/lib.rs (hunk 1 above)
Comment (Pagan-Idel) [reply src/lib.rs:863]: done

File: Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "pewpew"
-version = "0.5.12"
+version = "0.5.13"
```
[the remark this answers is not in the document]
Comment (Pagan-Idel) [reply Cargo.toml:3]: done

File: src/lib.rs [hunk 4]
```diff
@@ -873,6 +876,30 @@ fn create_try_run_future(
                 "RTT": "stats.rtt"
             }
         })"#
+    } else if matches!(try_config.format, TryRunFormat::Human) && matches!(try_config.skip_body_on, true) {
```
[the remark this answers is not in the document]
Comment (Pagan-Idel) [reply src/lib.rs:879]: Wouldn't we still have the same amount of if/then/elses for responses vs requests and human vs json?

[the remark this answers is not in the document]
Comment (tkmcmaster) [reply src/lib.rs:879]: Yes, but we wouldn't have the duplicated code where if we decide to change other parts of the output we have to change it in 2 or 4 spots.

File: src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkmcmaster) [reply src/lib.rs]: It looks like Clippy wants you to change it to `!try_config.skip_body_on` rather than checking to see if it `== false`

**Comment by Pagan-Idel** [comment]
@tkmcmaster made changes, let me know if that's more or less what you were looking for.

**Comment by tkmcmaster** [comment]
@Pagan-Idel It's looking good. Looks like rust fmt is failing now. There's a `pr.sh` script in the root of the project you can run to do all these checks locally. To fix the formatting automatically you can just run `cargo fmt --all` without the `--check`.

File: src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (tkmcmaster) [reply src/lib.rs]: Looks good. Thanks @Pagan-Idel

**Comment by Pagan-Idel** [comment]
We are still failing even after `cargo fmt --all`...

**Comment by tkmcmaster** [comment]
> We are still failing even after `cargo fmt --all`... Maybe it has something to do with the Spaces or Tab count?

Newer version of rust. Run `rustup update` to get the latest 1.72 and then rust fmt will work

**Review by tkmcmaster — APPROVED** [review]
Great change @Pagan-Idel Idel

**Pull request merged**